### PR TITLE
Cover degenerate shape + NaN/Inf reads on GPU and dask geotiff backends

### PR DESCRIPTION
**Pull request by brendancol**
## Summary

- The geotiff eager numpy path covers 1x1, 1xN, Nx1, all-NaN, Inf,
  and NaN-sentinel inputs end-to-end. The GPU, dask+numpy, and
  dask+cupy backends went unexercised for the same inputs.
- Adds 23 tests in `test_degenerate_shapes_backends_2026_05_11.py`
  pinning the contract across every non-eager backend.

Cat 3 HIGH (geometric edges):
- 1x1 single-pixel reads on dask+numpy, GPU, dask+cupy
- 1xN single-row reads on the same three backends
- Nx1 single-column reads on the same three backends
- 1x1, 1xN, Nx1 writes through `write_geotiff_gpu`

Cat 2 MEDIUM (NaN / Inf / nodata):
- All-NaN reads on GPU + dask+cupy
- Inf / -Inf reads on every non-eager backend
- NaN-sentinel mask on the float dask read path, including a
  sentinel block straddling a chunk boundary

No source changes; tests only. Test-coverage gap sweep 2026-05-11
pass 5 against the geotiff module.

## Test plan

- [x] `pytest xrspatial/geotiff/tests/test_degenerate_shapes_backends_2026_05_11.py` (23 pass on a GPU host)
- [x] Related geotiff tests still pass (`test_edge_cases.py`,
  `test_attrs_parity_1548.py`, `test_dask_int_nodata_chunks_1597.py`,
  `test_gpu_nodata_1542.py`, `test_dask_cupy_combined.py`).

### CODE REVIEW

File: xrspatial/geotiff/tests/test_degenerate_shapes_backends_2026_05_11.py [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+matching coverage. A regression that broke any backend for a 1-pixel
+row, 1-column strip, or all-NaN input would not surface until a user
+hit the production-traffic mosaic that triggered it.
+
+Test coverage gap sweep 2026-05-11 (pass 5) closes the Cat 3 (geometric
+edge case) and Cat 2 (NaN / Inf) gaps for the non-eager backends:
+
+* 1x1, 1xN, Nx1 reads on every backend (Cat 3 HIGH).
+* 1x1 and 1xN writes through ``write_geotiff_gpu`` (Cat 3 HIGH for
+  the GPU writer's degenerate-shape path).
```
[the remark this answers is not in the document]
Comment (brendancol) [reply xrspatial/geotiff/tests/test_degenerate_shapes_backends_2026_05_11.py]: Fixed in 8147b35: module docstring bullet now reads "1x1, 1xN, and Nx1 writes through write_geotiff_gpu" to match the actual GPU-writer coverage in the file.

File: xrspatial/geotiff/tests/test_degenerate_shapes_backends_2026_05_11.py [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+# ===========================================================================
+
+@_gpu_only
+class TestGpuWriterDegenerateShapes:
+    """``write_geotiff_gpu`` must accept 1-pixel and 1-row inputs.
+
+    The GPU writer's tile-encoding path uses an internal grid sizing
+    helper that fell back to host code for shapes smaller than the
+    default tile. The fallback exists but had no regression test that
+    would catch a future "fast-path only" refactor.
```
[the remark this answers is not in the document]
Comment (brendancol) [reply xrspatial/geotiff/tests/test_degenerate_shapes_backends_2026_05_11.py]: Fixed in 8147b35: TestGpuWriterDegenerateShapes docstring now says "must accept 1-pixel, 1-row, and 1-column inputs", and the section header above the class also lists Nx1.

File: xrspatial/geotiff/tests/test_degenerate_shapes_backends_2026_05_11.py [hunk 3]
```diff
@@ -0,0 +1,401 @@
...
+
+    def test_dask_numpy_chunks_smaller_than_sentinel_block(self, nan_sentinel_path):
+        """Sentinels split across two chunks still mask correctly.
+
+        The 2x2 sentinel block at rows 2-3 cols 2-3 lands in a single
+        chunk for chunks=4 (rows 0-3) but straddles a chunk boundary
+        for chunks=2 (rows 2-3 split between chunks 1 and 2). This
+        exercises the per-block sentinel comparison.
+        """
+        path, _ = nan_sentinel_path
```
[the remark this answers is not in the document]
Comment (brendancol) [reply xrspatial/geotiff/tests/test_degenerate_shapes_backends_2026_05_11.py:398]: Fixed in 8147b35: class docstring now reads "Float raster with a finite nodata sentinel (-9999.0) is masked to NaN consistently across backends on read", matching what the fixture actually writes.
